debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -530,13 +531,22 @@ protected void execute() {
                         schema.applyDdl(source, dbName, "DROP DATABASE IF EXISTS " + quote(dbName), this::enqueueSchemaChanges);
 
                         final StringBuilder createDatabaseDddl = new StringBuilder("CREATE DATABASE " + quote(dbName));
-                        final String defaultDatabaseCharset = databaseCharsets.get(dbName);
-                        if (defaultDatabaseCharset != null) {
-                            logger.debug("Setting default charset '{}' for database '{}'", defaultDatabaseCharset, dbName);
-                            createDatabaseDddl.append(" CHARSET ").append(defaultDatabaseCharset);
-                        }
-                        else {
-                            logger.info("Default database charset for '{}' not found", dbName);
+                        final DatabaseLocales defaultDatabaseLocales = databaseCharsets.get(dbName);
+                        if (defaultDatabaseLocales != null) {
+                            if (defaultDatabaseLocales.charset != null) {
+                                logger.debug("Setting default charset '{}' for database '{}'", defaultDatabaseLocales.charset, dbName);
+                                createDatabaseDddl.append(" CHARSET ").append(defaultDatabaseLocales.charset);
+                            }
+                            else {
+                                logger.info("Default database charset for '{}' not found", dbName);
+                            }
+                            if (defaultDatabaseLocales.collation != null) {
+                                logger.debug("Setting default collation '{}' for database '{}'", defaultDatabaseLocales.collation, dbName);
+                                createDatabaseDddl.append(" COLLATE ").append(defaultDatabaseLocales.collation);
+                            }
+                            else {
+                                logger.info("Default database collation for '{}' not found", dbName);
+                            }
```

Comment:
How about pushing all this into `DatabaseLocales`:

    defaultDatabaseLocales.appendTo(createDatabaseDddl);

And it would take care of all the conditional logic internally. Keeps things easier here and should make it simpler to integrate into the new impl, too.

Reply:
@gunnarmorling Done